### ISSUE TITLE
hg19 -> hg38 automatic liftover problems

### ISSUE THREAD

**Issue by EugeneEA** [KarchinLab/open-cravat#35]
Hi, I've noticed some strange thing in the liftover results

my input vcf file has following line:
chr10   48414222        rs556004917     G       A  

For reasons unknow, with the command ` oc run test.vcf -l hg19 --skip reporter -a dbsnp`  I'm getting the output with hg38 coordinates: 47325138 (see the attach figure)

![123](https://user-images.githubusercontent.com/57629300/90507260-4b9d5900-e15e-11ea-8f22-d409764401b9.png)

But it has to be 47325140 (even alt/ref now incorrect!) ! I'm not sure what is going on here and would appriciate any advice! 
PS UCSC liftover tools (https://genome.ucsc.edu/cgi-bin/hgLiftOver) also convert 48414222->47325140

Best, Eugene

**Comment by rkiminsilico** [comment]
Hi Eugene, thanks for letting us know. We'll take a look.

**Comment by rkiminsilico** [comment]
The issue was due to how OpenCRAVAT was handling liftover for the regions which were inverted between hg19 and hg38. This fix will be included in our next release of OpenCRAVAT. If you want to try this fix now, a workaround is:
```
1. Find and download cravat_convert.py in 2.1.1 branch of this repo.
2. Put the cravat_convert.py into your OpenCRAVAT pip package directory (which can be found with "pip list -v|grep open-cravat". Add "cravat" to the found site-packages directory).
```

**Comment by rkiminsilico** [comment]
@EugeneEA Yes, OpenCRAVAT uses pyliftover internally and pyliftover uses 0-based coordinates while OpenCRAVAT uses 1-based coordinates. How OpenCRAVAT handled the communication between the two was the cause of the issue related to the genomic regions which were inverted from hg19 to hg38.

**Comment by EugeneEA** [comment]
Hi, I'veagain noticed some problems with liftover procedure:
here is the results of hg19 mapped vcf, submitted to the oc (version 2.2.3) with -l hg19 parameter, notice the absence of the gnomad3 annotation:
![image](https://user-images.githubusercontent.com/57629300/109797142-1bb13100-7c2a-11eb-97f4-e1ab1d6444f4.png)

and here is the same vcf file, which was lifted over to hg38 with gatk picard tool, as you can see now the gnomad3 annotation os present:
![image](https://user-images.githubusercontent.com/57629300/109797371-6df25200-7c2a-11eb-87ea-d9ef05b27ea2.png)

Basically cravat maneges to liftover the position but failed to swap alleles (the ref allele was changed between hg19 to hg38), therefore new SNP became unic in the hg38 and absent from gnomad. Could you, please, comment on this bug? Should I just routinely run picard before cravat? (btw why you do not use pickard internally?)

Best, Eugene

**Comment by rkimoakbioinformatics** [comment]
@EugeneEA Thanks for finding the bug and sorry for the inconvenience. A fix will be included in the next release of OpenCRAVAT. Meanwhile, you can try the fix as follows.
```
1. pip list -v|grep open-cravat
   (This will show you where OpenCRAVAT is installed.)
2. Overwrite cravat_convert.py in the folder found in (2) with the attached cravat_convert.py.
```
Let me know if works for you.

We didn't use Picard because we wanted to keep minimal dependencies and to use only Python dependencies.

**Comment by EugeneEA** [comment]
@rkimoakbioinformatics thanks a lot for such quick reply! It did work out,
 best, Eugene

**Comment by rkimoakbioinformatics** [comment]
@EugeneEA That's great. Glad it worked.

**Comment by EugeneEA** [comment]
Hi, I've noticed that you have not change the file in your latest version? was it by mistake or are there any meaningfull reasons? 
Best, Eugene

**Comment by rkimoakbioinformatics** [comment]
@EugeneEA If you mean the fix at https://github.com/KarchinLab/open-cravat/issues/35#issuecomment-789755333, this fix came after the latest version, open-cravat 2.2.3. We are close to releasing the next version, open-cravat 2.2.4, and this version will have the fixed cravat_convert.py file.

**Comment by rkimoakbioinformatics** [comment]
@EugeneEA I hope that OpenCRAVAT is working for you well. We are conducting interviews to learn what else we could do for genomic variant researchers. Would you be available for such an interview? It will take just about 15 minutes and your input will be invaluable to make OpenCRAVAT even better. It does not have to be about your experience with OpenCRAVAT. We just would like to hear what your experience has been with genomic variant analysis tools. If so, please let me know at rkim@oakbioinformatics.com.
